### DISCUSSION TITLE
server and client failed to  connect to each other in different docker container

### DISCUSSION THREAD

**Discussion by Allennnn2022** [alibaba/xquic#425] [Q&A]
### What happened?

A clear and concise description of what the bug is. I run test_server with：/test_server -a 172.21.112.1 -p 4488   -l e > /dev/null & and test_client with: test_client -a 127.0.0.1 -p 4488 -s 1024000 -l e -E in different docker container, however client fails to connect to the server with error message： 
![image](https://github.com/alibaba/xquic/assets/114003920/5afc3beb-9eec-45fe-b622-07be4601a636)
 and catch no package in wireshark If I run client with other server ip, it seems to work but still no response, then connection close ![image](https://github.com/alibaba/xquic/assets/114003920/8f84e6e6-5136-4c36-be49-e37d7c1a6dbd) 
differently, I can catch quic packages in wireshark What 's the problem?

### Steps To Reproduce

Information and Steps to reproduce the behavior.

**Comment by Kulsk** [comment]
when use -a option on test_server, it means test_server will bind fd on that address. 
![image](https://github.com/alibaba/xquic/assets/22413114/8ca0ac86-b4a9-48e5-a5f2-70825ae45f98)

what is the interface and address of the server container? make sure port 4488 is allowed and the interface is right
